### PR TITLE
Add accelerate docs

### PR DESCRIPTION
**Pull request by htahir1**
## Describe changes
I added some accelerate docs

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/how-to/training-with-gpus/accelerate-distributed-training.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+---
+description: Run distributed training with Hugging Face's Accelerate library in ZenML pipelines.
+---
+
+# Distributed training with 🤗 Accelerate
+
+There are several reasons why you might want to scale your machine learning pipelines to utilize distributed training, such as leveraging multiple GPUs or training across multiple nodes. ZenML now integrates with Hugging Face's Accelerate library to make this process seamless and efficient.
```
Comment (strickvl) [inline docs/book/how-to/training-with-gpus/accelerate-distributed-training.md:7]:
```suggestion
There are several reasons why you might want to scale your machine learning pipelines to utilize distributed training, such as leveraging multiple GPUs or training across multiple nodes. ZenML now integrates with [Hugging Face's Accelerate library](https://github.com/huggingface/accelerate) to make this process seamless and efficient.
```

File: docs/book/how-to/training-with-gpus/accelerate-distributed-training.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+### Configuration
+
+The `run_with_accelerate` function accepts various arguments to configure your distributed training environment. Some common arguments include:
+
+- `num_processes`: The number of processes to use for distributed training.
+- `cpu`: Whether to force training on CPU.
+- `multi_gpu`: Whether to launch distributed GPU training.
+- `mixed_precision`: Mixed precision training mode ('no', 'fp16', or 'bf16').
+
+### Important considerations
```
Comment (strickvl) [inline docs/book/how-to/training-with-gpus/accelerate-distributed-training.md:42]:
```suggestion
### Important Usage Notes
```

File: docs/book/how-to/training-with-gpus/accelerate-distributed-training.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+1. The `run_with_accelerate` function cannot be used directly on steps using the '@' syntax. Use it within your pipeline definition instead.
+
+2. Steps defined inside the entrypoint script cannot be used with `run_with_accelerate`. Move your step code to another file and import it.
+
+3. Accelerated steps do not support positional arguments. Use keyword arguments when calling your steps.
+
+4. If `run_with_accelerate` is misused, it will raise a `RuntimeError` with a helpful message explaining the correct usage.
+
```
Comment (strickvl) [inline docs/book/how-to/training-with-gpus/accelerate-distributed-training.md:51]:
```suggestion

{% hint style="info" %}
To see a full example where Accelerate is used within a ZenML pipeline, check out our <a href="https://github.com/zenml-io/zenml-projects/blob/main/llm-lora-finetuning/README.md">llm-lora-finetuning</a> project which leverages the distributed training functionalities while finetuning an LLM.
{% endhint %}

```

**Review by strickvl — CHANGES_REQUESTED** [review]

**Review by avishniakov — APPROVED** [review]

File: docs/book/how-to/training-with-gpus/accelerate-distributed-training.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+@pipeline
+def training_pipeline(num_processes: int):
+    run_with_accelerate(training_step, num_processes=num_processes)()
+```
+
+The `run_with_accelerate` function wraps your step, enabling it to run with Accelerate's distributed training capabilities. It accepts various arguments that correspond to Accelerate CLI options.
+
+{% hint style="info" %}
+For a complete list of available arguments and more details, refer to the [Accelerate CLI documentation](https://huggingface.co/docs/accelerate/en/package_reference/cli#accelerate-launch).
```
[the remark this answers is not in the document]
Comment (htahir1) [reply docs/book/how-to/training-with-gpus/accelerate-distributed-training.md:30]: @strickvl pretty sure it works . lets merge to develop first and see

**Review by strickvl — APPROVED** [review]

**Pull request merged**